client/proposal.go

```diff
@@ -371,15 +371,6 @@ func (c *Client) validChannelProposalAcc(
 		return errors.Errorf("mismatched proposal ID %b and accept ID %b", propID, accID)
 	}
 
-	if proposal.Type() == wire.SubChannelProposal {
```

Comment:
This is already checked at `validTwoPartyProposal`, where it also fits much better.

Reply:
And it can not happen that the parent channel is closed between the `propose` and `proposeAccept` message?

Reply:
You mean via `Channel.Close()`? I guess that could happen, but it would also be caught when instantiating the channel.

In general, I think we don't do us a favor if we do a large number of checks all over the code. It makes the code harder to maintain and I don't see many practical benefits. Instead, we should be more conscious about which properties we want to ensure and then place a concise set of checks at selected locations.

So the question here is: What are you trying to achieve? Is it that you want to ensure that an open child channel must not have a closed parent channel? Once we have the goal fixed, we can argue which checks are suitable and sufficient. By the way, I don't think there is anything that prevents one from closing a parent channel at any time. This would be probably worth addressing, in another issue.

Reply:
Ok thanks for the reply. I see it similar, but it seems like a time-intensive task…